src/pages/Search/index.tsx

```diff
@@ -10,28 +11,53 @@ import useSearchingMode from './hooks/useSearchingMode';
 
 const useSearchForm = () => {
   const [text, setText] = useState('');
+  const [entered, setEntered] = useState(false);
+  const [submittedText, setSubmittedText] = useState('');
   const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
     setText(e.target.value);
+    setEntered(false);
   };
 
+  const handleEnter = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === 'Enter' && text) {
+      setSubmittedText(text);
+      setEntered(true);
+    }
+  };
+
+  useEffect(() => {
+    setEntered(false);
+  }, [text]);
+
   return {
     text,
     handleChange,
+    handleEnter,
+    submittedText,
+    entered,
   };
 };
 
 export default function Search(): JSX.Element {
-  const { text, handleChange } = useSearchForm();
+  const {
+    text, handleChange, handleEnter, entered,
+  } = useSearchForm();
   const isSearching = useSearchingMode();
   const { isMobile } = useMediaQuery();
+
   return (
     <div>
       <div className={styles.search}>
         <section>
           {isMobile && <NavigationBar />}
-          {isMobile ? !isSearching && <Recommendation /> : <Recommendation />}
-          <SearchInput onChange={handleChange} text={text} />
-          {!isMobile && isSearching ? <RelatedSearches text={text} /> : <RollingBanner />}
+          {!entered && (
+            (isMobile && !isSearching) || !isMobile
+          ) && <Recommendation />}
+          {!isMobile && !entered
+            ? <SearchInput onChange={handleChange} text={text} onKeyDown={handleEnter} />
+            : <SearchDetails />}
+          {!isMobile && isSearching && !entered && <RelatedSearches text={text} />}
+          {!isMobile && !isSearching && !entered && <RollingBanner />}
```

Comment:
아래의 코드도 복잡해 보이려나요...?
```suggestion
          {!entered && (
              !isMobile && isSearching ? <RelatedSearches text={text} /> : <RollingBanner />
)}
```

Reply:
훨 좋은것 같습니다! 코드에서 삼항연산자를 이중으로 사용하면 오류가 발생할 때가 있던데.. 위의 코드는 잘 돌아가네요. 
혹시 기준을 아시나요?